cartography/intel/aws/rds.py

```diff
@@ -200,17 +200,23 @@ def _attach_read_replicas(neo4j_session, read_replicas, aws_update_tag):
     Attach read replicas to their source instances
     """
     attach_replica_to_source = """
-    MATCH (replica:RDSInstance{id:{ReplicaArn}}),
-    (source:RDSInstance{db_instance_identifier:{SourceInstanceIdentifier}})
+    MATCH (replica:RDSInstance{id:{ReplicaArn}})
+    MERGE (source:RDSInstance{id:{SourceArn}})
     MERGE (replica)-[r:IS_READ_REPLICA_OF]->(source)
     ON CREATE SET r.firstseen = timestamp()
     SET r.lastupdated = {aws_update_tag}
     """
     for replica in read_replicas:
+        # NOTE: AWS returns DBInstanceIdentifier for intra-region replicas
+        # but returns ARN for inter-region replicas
+        source_arn = replica['ReadReplicaSourceDBInstanceIdentifier']
```

Comment:
I recommend extracting this data adjustment to its own transformation function so that we separate transforming and loading data. 

It would look like this:
```python
def _transform_read_replicas(read_replicas):
    for replica in read_replicas:
      source_arn = replica['ReadReplicaSourceDBInstanceIdentifier']
      if not source_arn.startswith('arn'):
          # Take the replica's ARN and chop off the replica's DBInstanceIdentifier so we can use it as a prefix
          arn_prefix = ":".join(replica['DBInstanceArn'].split(":")[:-1])
          source_arn = arn_prefix + source_arn
          replica['source_arn'] = source_arn
    return read_replicas
```

Then at the bottom of `load_read_replicas()`, we would call 

```python
_transform_read_replicas(read_replicas) # should modify the variable since python is pass by name, double check this though
_attach_read_replicas(neo4j_session, read_replicas, aws_update_tag)
```

And then finally `_attach_read_replicas()` would be the way you have it except with 
```python
def _attach_read_replicas(neo4j_session, read_replicas, aws_update_tag):
. . .
    neo4j_session.run(
        attach_replica_to_source,
        ReplicaArn=replica['DBInstanceArn'],
        SourceArn=replica['source_arn'],
. . .
```